.github/workflows/coverage-check.yml

```diff
@@ -0,0 +1,79 @@
+name: Coverage Check
+
+on:
+  pull_request:
+    branches: [ main ]
```

Comment:
By the way, I find `branches: [ "**" ]` more useful.

See example https://github.com/awslabs/package-validation-tool/blob/7413800c07495f670604e92cc8a9461f47b732b0/.github/workflows/basic-ci.yml#L7